conseil-common/src/main/scala/tech/cryptonomic/conseil/common/bitcoin/BitcoinPersistence.scala

```diff
@@ -0,0 +1,131 @@
+package tech.cryptonomic.conseil.common.bitcoin
+
+import java.sql.Timestamp
+
+import cats.Id
+import cats.effect.{Concurrent, Resource}
+import com.typesafe.scalalogging.LazyLogging
+import slick.jdbc.PostgresProfile.api._
+
+import tech.cryptonomic.conseil.common.util.Conversion
+import tech.cryptonomic.conseil.common.util.Conversion.Syntax._
+import tech.cryptonomic.conseil.common.bitcoin.rpc.json.{Block, Transaction, TransactionInput, TransactionOutput}
+import tech.cryptonomic.conseil.common.bitcoin.BitcoinPersistence._
+
+/**
+  * Bitcoin persistence into the database using Slick.
+  */
+class BitcoinPersistence[F[_]: Concurrent] extends LazyLogging {
+
+  /**
+    * Create [[DBIO]] seq with block (with transactions, inputs and outputs) that can be wrap into one transaction.
+    *
+    * @param block JSON_RPC block
+    * @param transactions JSON_RPC block's transactions
+    */
+  def createBlock(
+      block: Block,
+      transactions: List[Transaction]
+  ): DBIOAction[Unit, NoStream, Effect.Write] =
+    DBIO.seq(
+      Tables.Blocks += block.convertTo[Tables.BlocksRow],
+      Tables.Transactions ++= transactions.map(_.convertTo[Tables.TransactionsRow]),
+      Tables.Inputs ++= transactions.map(_.vin).flatten.map(_.convertTo[Tables.InputsRow]),
+      Tables.Outputs ++= transactions.map(_.vout).flatten.map(_.convertTo[Tables.OutputsRow])
+    )
+}
+
+object BitcoinPersistence {
+
+  /**
+    * Create [[cats.Resource]] with [[BitcoinPersistence]].
+    */
+  def resource[F[_]: Concurrent]: Resource[F, BitcoinPersistence[F]] =
+    for {
```

Comment:
why not simply build a pure resource value as in
`Resource.pure(new BitcoinPersistence)` ?

Reply:
sorry, there was a dependency to `tx` before